internal/tracing/handlers/handler.go

```diff
@@ -43,6 +44,17 @@ func BuildOptions(ctx context.Context, k8s client.Client, mcAddon *addonapiv1alp
 	opts.OpenTelemetryCollector = otelCol
 	klog.Info("OpenTelemetry Collector template found")
 
+	if userWorkloads.InstrumentationEnabled {
+		klog.Info("Retrieving Instrumentation template")
+		keys = addon.GetObjectKeys(mcAddon.Status.ConfigReferences, otelv1beta1.GroupVersion.Group, addon.InstrumentationResource)
+		instr := &otelv1alpha1.Instrumentation{}
+		if err := k8s.Get(ctx, keys[0], instr, &client.GetOptions{}); err != nil {
```

Comment:
Here we should replicate what we have in https://github.com/rhobs/multicluster-observability-addon/blob/e7edf8b302ae2438feb33caf7374b7d4babeb76f/internal/tracing/handlers/handler.go#L32-L39 since in the future users will be able to provide multiple instrumentation resources